api/v1/verticaautoscaler_webhook.go

```diff
@@ -152,6 +176,99 @@ func (v *VerticaAutoscaler) validateCustomAutoscaler(allErrs field.ErrorList) fi
+		if v.Spec.CustomAutoscaler.ScaledObject == nil && (v.Spec.CustomAutoscaler.Type == ScaledObject || v.Spec.CustomAutoscaler.Type == "") {
+			err := field.Invalid(pathPrefix.Child("type"),
+				v.Spec.CustomAutoscaler.Type,
+				fmt.Sprintf("customAutoscaler.ScaledObject must be non-nil if customAutoscaler.type is %s or empty.", ScaledObject),
+			)
+			allErrs = append(allErrs, err)
+		}
+	}
+	return allErrs
+}
+
+// validateReplicas will check if minReplicas and maxReplicas are valid
+func (v *VerticaAutoscaler) validateReplicas(allErrs field.ErrorList) field.ErrorList {
+	if v.Spec.CustomAutoscaler != nil {
+		if v.Spec.CustomAutoscaler.Type == HPA {
+			pathPrefix := field.NewPath("spec").Child("customAutoscaler").Child("HPA")
+			if v.Spec.CustomAutoscaler.Hpa != nil {
+				if v.Spec.CustomAutoscaler.Hpa.MaxReplicas == 0 {
+					err := field.Invalid(pathPrefix.Child("MaxReplicas"),
+						v.Spec.CustomAutoscaler.Hpa.MaxReplicas,
+						"maxReplicas must be set.")
+					allErrs = append(allErrs, err)
+				}
+
+				if v.Spec.CustomAutoscaler.Hpa.MinReplicas != nil &&
+					v.Spec.CustomAutoscaler.Hpa.MaxReplicas < *v.Spec.CustomAutoscaler.Hpa.MinReplicas {
+					err := field.Invalid(pathPrefix.Child("MaxReplicas"),
+						v.Spec.CustomAutoscaler.Hpa.MaxReplicas,
+						fmt.Sprintf("maxReplicas %d cannot be less than minReplicas %d.",
+							v.Spec.CustomAutoscaler.Hpa.MaxReplicas, v.Spec.CustomAutoscaler.Hpa.MinReplicas),
+					)
+					allErrs = append(allErrs, err)
+				}
+			}
+		}
+
+		if v.Spec.CustomAutoscaler.Type == ScaledObject {
+			pathPrefix := field.NewPath("spec").Child("customAutoscaler").Child("ScaledObject")
+			if v.Spec.CustomAutoscaler.ScaledObject != nil {
+				if v.Spec.CustomAutoscaler.ScaledObject.MaxReplicas == nil {
+					err := field.Invalid(pathPrefix.Child("MaxReplicas"),
+						v.Spec.CustomAutoscaler.ScaledObject.MaxReplicas,
+						"maxReplicas must be set.")
+					allErrs = append(allErrs, err)
+				}
+
+				if v.Spec.CustomAutoscaler.ScaledObject.MinReplicas != nil && v.Spec.CustomAutoscaler.ScaledObject.MaxReplicas != nil &&
+					*v.Spec.CustomAutoscaler.ScaledObject.MaxReplicas < *v.Spec.CustomAutoscaler.ScaledObject.MinReplicas {
+					err := field.Invalid(pathPrefix.Child("MaxReplicas"),
+						v.Spec.CustomAutoscaler.ScaledObject.MaxReplicas,
+						fmt.Sprintf("maxReplicas %d cannot be less than minReplicas %d.",
+							v.Spec.CustomAutoscaler.ScaledObject.MaxReplicas, v.Spec.CustomAutoscaler.ScaledObject.MinReplicas),
+					)
+					allErrs = append(allErrs, err)
+				}
+			}
+		}
+	}
+	return allErrs
+}
+
+// validateMetricsName makes sure 2 metrics cannot have the same name
+func (v *VerticaAutoscaler) validateMetricsName(allErrs field.ErrorList) field.ErrorList {
+	metrics := []string{}
+	pathPrefix := field.NewPath("spec").Child("customAutoscaler")
+	if v.Spec.CustomAutoscaler != nil && v.Spec.CustomAutoscaler.Type == ScaledObject {
+		if v.Spec.CustomAutoscaler.ScaledObject != nil {
```

Comment:
I suggest you use IsCustomMetricsEnabled(), IsHpaEnabled(), IsScaledObjectEnabled()